modules/library/metadata/pom.xml

```diff
@@ -130,6 +130,10 @@
       <scope>provided</scope>  <!-- Not required for default execution. -->
       <optional>true</optional>
     </dependency>
+    <dependency>
+        <groupId>com.google.guava</groupId>
+        <artifactId>guava</artifactId>
+    </dependency>
```

Comment:
I am a bit uncomfortable with the use of guava as it continues to produces incompatibilities with downstream applications. Most of the benefits we were seeking from guava are now available as part of Java.

What specific features of Guava are you looking to leverage here?

Reply:
Agreed, having Guava as a core dependency is likely going to be a blocker. We have core devs trying to promote a "un-guava-ifycation" of our code base (not as easy as it seems, but I agree we should not have it everywhere).

Reply:
I used it to filter an Iterator and for an `ImmutableMap`. Removed that and thus the dependency in aa015e8.